config/scripts/check.sh

```
@@ -1105,232 +1439,1219 @@ check_disk_hotplug() {
+                        local ps_generate
+                        ps_generate=''
+                        ps_generate+='$ex=@("C","D"); '
+                        ps_generate+='$vols=Get-Volume|Where-Object{$_.DriveLetter -and $_.DriveType -eq "Fixed" -and $_.DriveLetter -notin $ex}; '
+                        ps_generate+='if($vols.Count -eq 0){"DATAGEN_NO_TARGET_DRIVES"; exit 1} '
+                        ps_generate+='$allExist=$true; foreach($v in $vols){$e=@(Get-ChildItem -Path "$($v.DriveLetter):\" -Directory -Filter "fio_data_dir_*" -EA SilentlyContinue); '
+                        ps_generate+="if(\$e.Count -ne ${dir_count}){\$allExist=\$false; break}} "
+                        ps_generate+='if($allExist){"DATAGEN_SKIPPED data already exists"; exit 0} '
+                        ps_generate+='"DATAGEN_STARTING drives=$(($vols.DriveLetter)-join ",")"; '
+                        ps_generate+='$d=[char]36; '
+                        if [ "${depth_count}" -le 1 ]; then
+                            ps_generate+='$fn="fio_data_dir_${d}jobnum\bench_file_${d}filenum.dat"; '
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)